src/plugins/agentDesktop/agentdesktop.ts

```diff
@@ -13,38 +15,95 @@ class AgentDesktopPlugin {
 
     }
     onHostCreate() {
-        let me:any = this;
+        let me: any = this;
         let cwInstance = me.hostInstance;
 
-        cwInstance.on("jwtGrantSuccess", (response:any) => {
+        cwInstance.on("jwtGrantSuccess", (response: any) => {
             if (!this.agentDesktopInfo) {
                 me.onInit();
-                this.config=me.extend(me,response)
+                this.config = me.extend(me, response)
                 //me.AgentDesktop(response.userInfo.userId, response);
                 this.agentDesktopInfo = new AgentDesktopPluginScript(this.config);
             }
         });
 
     }
     onInit() {
-        // let me = this;
+        let me: any = this;
         this.appendVideoAudioElemnts()
-        // this.koreCoBrowse = me.koreCoBrowseInit({});
-        // this.rrweb = me.rrwebInit({});
+
+        // agent connected and disconnected events
+        me.hostInstance.on('onWSMessage', (event: any) => {
+            if (event.messageData?.message?.type === 'agent_connected') {
+                localStorage.setItem("isAgentConnected", "true")
+            } else if (event.messageData?.message?.type === 'agent_disconnected') {
+                localStorage.setItem("isAgentConnected", "false")
+            }
+        });
+
+        // sent event style setting in user chat 
+        me.hostInstance.on('afterRenderMessage', (event: any) => {
+
+            if (localStorage.getItem("isAgentConnected") != "true") return;
```

Comment:
@KambagiriswamyBo-Kore 
do we really need localstorage? avoid if possible otherwise use the key with kore namespacing to avoid collision
Ex:
kr-agent-status

Reply:
@RajasekharBa-Kore We need this flag to check if agent is connected or not even on page refresh. Changed the variable name as suggested.